packages/twenty-front/src/modules/ui/input/components/internal/date/components/InternalDatePicker.tsx

```diff
@@ -454,6 +454,14 @@ export const DateTimePicker = ({
     }
   };
 
+  useEffect(() => {
+    const month = dateParsed.get('month');
```

Comment:
Please find another pattern without a useEffect, it's a bad practice to synchronize state with useEffect, and in that case we can certainly avoid it. 

You can do it by placing setDatePickerMonth and setDatePickerYear in the right handlers.

Reply:
Please let's not change the handler strategy because there are many moving parts and this setup is working so your change has a high risk of introducing regression. Let's only add calls to setDatePickerMonth and setDatePickerYear where needed.